tdrs-frontend/src/components/Reports/Reports.jsx

```diff
@@ -27,61 +29,191 @@ function Reports() {
+  const [touched, setTouched] = useState({})
+
+  const quarters = {
+    Q1: 'Quarter 1 (October - December)',
+    Q2: 'Quarter 2 (January - March)',
+    Q3: 'Quarter 3 (April - June)',
+    Q4: 'Quarter 4 (July - September)',
   }
 
   const handleSearch = () => {
-    setIsToggled(true)
+    if (!isUploadReportToggled) {
+      // Clear previous errors
+      setFormValidationState({})
+      // Filter out non-truthy values
+      const form = [selectedYear, selectedStt, selectedQuarter].filter(Boolean)
+
+      if (form.length === 3) {
+        setIsToggled(true)
+      } else {
+        // create error state
+        setFormValidationState({
+          year: !selectedYear,
+          stt: !selectedStt,
+          quarter: !selectedQuarter,
+          errors: 3 - form.length,
+        })
+        setTouched({
+          year: true,
+          stt: true,
+          quarter: true,
+        })
+      }
+    }
+  }
+
+  const selectYear = ({ target: { value } }) => {
+    setIsToggled(false)
+    dispatch(setYear(value))
+    setTouched((currentForm) => ({ ...currentForm, year: true }))
   }
 
+  const selectQuarter = ({ target: { value } }) => {
+    setIsToggled(false)
+    dispatch(setQuarter(value))
+    setTouched((currentForm) => ({ ...currentForm, quarter: true }))
+  }
   // Non-OFA Admin users will be unable to select an STT
   // prefer => `auth.user.stt`
 
   const selectStt = (value) => {
+    setIsToggled(false)
     dispatch(setStt(value))
+    setTouched((currentForm) => ({ ...currentForm, stt: true }))
   }
 
+  useEffect(() => {
+    if (!isUploadReportToggled) {
+      const form = [selectedYear, selectedStt, selectedQuarter].filter(Boolean)
+      const touchedFields = Object.keys(touched).length
+
+      const errors = touchedFields === 3 ? 3 - form.length : 0
+
+      setFormValidationState((currentState) => ({
+        ...currentState,
+        year: touched.year && !selectedYear,
+        stt: touched.stt && !selectedStt,
+        quarter: touched.quarter && !selectedQuarter,
+        errors,
+      }))
+    }
+  }, [
+    isUploadReportToggled,
+    selectedYear,
+    selectedStt,
+    selectedQuarter,
+    setFormValidationState,
+    touched,
+  ])
+
   const reportHeader = `${
-    selectedStt
-      ? `${
-          sttList.find((stt) => stt.name.toLowerCase() === selectedStt).name
-        } - `
-      : ''
-  }Fiscal Year ${selectedYear}`
+    sttList?.find((stt) => stt?.name?.toLowerCase() === selectedStt)?.name
+  } - Fiscal Year ${selectedYear} - ${quarters[selectedQuarter]}`
+
+  const errorsCount = formValidation.errors
 
   return (
     <>
       <div className={classNames({ 'border-bottom': isUploadReportToggled })}>
+        {Boolean(formValidation.errors) && (
+          <div className="margin-top-4 usa-error-message" role="alert">
+            There {errorsCount === 1 ? 'is' : 'are'} {formValidation.errors}{' '}
+            errors in this form
```

Comment:
```suggestion
            error(s) in this form
```

Reply:
this is minor and im not sure how parentheses might be treated 😄

Reply:
Nice catch